### DISCUSSION TITLE
buyerIdentity seems to rely on 'accepts marketing'

### DISCUSSION THREAD

**Discussion by justinhenricks** [Shopify/function-examples#508] [Feedback: Payment Customization API]
We noticed some strange behavior today where for customers that have not accepted marketing, `buyerIdentity` seems to be coming in as null whereas customers who have accepted marketing, `buyerIdentity` comes in as expected.

Here is the query for buyerIdentity we are using:

```
buyerIdentity {
      email
      customer {
          amountSpent {
              amount
          }
          hasTags(tags: $customerTags) {
              hasTag
              tag
          }
          numberOfOrders
      }
      isAuthenticated
      purchasingCompany {
          company {
              id
          }
      }
  }
```

Here are screenshots of two customers, one accepting marketing and one not as well as the corresponding input I am seeing when the function runs.

**accepts marketing:**
<img width="1117" alt="Screenshot 2024-05-29 at 2 43 15 PM" src="https://github.com/Shopify/function-examples/assets/3218907/b47034e7-da6b-4e2d-8f89-d206eb8844a8">
<img width="921" alt="Screenshot 2024-05-29 at 2 43 10 PM" src="https://github.com/Shopify/function-examples/assets/3218907/8403b452-bc4f-4d0b-9277-7cce91f73518">

**does not accept marketing:**
<img width="1092" alt="Screenshot 2024-05-29 at 2 43 21 PM" src="https://github.com/Shopify/function-examples/assets/3218907/7f7cc60d-a611-48a5-a251-d813f5318d84">
<img width="460" alt="Screenshot 2024-05-29 at 2 43 35 PM" src="https://github.com/Shopify/function-examples/assets/3218907/3e4f7693-31bc-41c5-b8df-3143efa7e806">

It seems strange that accepts marketing would effect this.. is it possible it is because we are asking for email in our query?

**Comment by tjoyal** [comment]
After investigation, we pinpointed a scenario where the buyer identity was not consistently loaded during the initial page render. 

This problem was confined to the initial page render and would have been automatically corrected once the page finished loading.

We have addressed and resolved this issue.
